commerce_coordinator/apps/titan/pipeline.py

```diff
@@ -60,14 +68,39 @@ def run_filter(self, edx_lms_user_id, payment_number=None, **kwargs):  # pylint:
         try:
             payment = api_client.get_payment(
                 edx_lms_user_id=edx_lms_user_id,
-                payment_number=payment_number
             )
         except HTTPError as exc:
             logger.exception("[GetTitanPayment] Payment %s not found for user: %s", payment_number, edx_lms_user_id)
             raise PaymentNotFound from exc
         payment_serializer = PaymentSerializer(data=payment)
         payment_serializer.is_valid(raise_exception=True)
-        return payment_serializer.data
+        payment = payment_serializer.data
+
+        order_uuid = kwargs.get('order_uuid')
+        if order_uuid and payment['order_uuid'] != kwargs['order_uuid']:
+            raise InvalidOrderPayment(
+                f'Requested order_uuid "{order_uuid}" does not match with '
+                f'order_uuid "{payment["order_uuid"]}" in Spree system.'
+            )
+
+        if payment_number and payment['payment_number'] != payment_number:
+            raise PaymentMismatch(
+                f'Requested payment number "{payment_number}" does not match with '
+                f'payment number "{payment["payment_number"]}" in Spree system.'
+            )
+
+        if kwargs.get('validate_payment_processing_state'):
+            # make sure fetch payment is processable.
+            if payment['state'] == PaymentState.COMPLETED.value:
+                raise AlreadyPaid(f'Requested payment "{payment_number}" for processing is already paid.')
+            if payment['state'] == PaymentState.PENDING.value:
+                raise ProcessingAlreadyRequested(
+                    f'Requested payment "{payment_number}" for processing is already processing.'
+                )
```

Comment:
This `GetTitanPayment` pipeline step is starting to take on a lot of possible use cases:

1. Get the latest payment for a user
    - For Mark Payment as Ready for Processing
2. Get a specific payment for a user
    - (Removed in this PR - will be needed later as Payment MFE polls Coordinator for Get Payment State)  
3. Check the latest payment for a user matches an `order_uuid` and `payment_number`
    - For Mark Payment as Ready for Processing
4. Check a specific payment for a user matches an `order_uuid`.
    - For Get Payment State
5. Check the latest or a specific payment for a user is in the DRAFT state.
    - Via `validate_payment_processing_state`
    - For Mark Payment as Ready for Processing

Would it make sense to split this `GetTitanPayment` pipeline steps into multiple pipeline steps (like `CheckTitanPaymentInDraftState`, `CheckLatestTitanPaymentIsExpected`), which different actions could call depending on their needs?

If it does make sense, don't want to blow up the scope of this PR - maybe this is good for a follow-on ticket.

Reply:
Yes, @pshiu Thank you for looking into it in this depth. You were right about the fact this step was doing a lot of things. So I have created a new step `ValidatePaymentReadyForProcessing` for all processing related to Ready for Processing. 
I think we are ok without CheckLatestTitanPaymentIsExpected as these checks are common to every use case.